### PR TITLE
No breaking changes for `Role::Set` + Fixes

### PR DESCRIPTION
**Pull request by IRacle1**
- Warhead fix
- Scp106Role::UsePortal fix

### CODE REVIEW

File: Exiled.API/Features/Roles/Scp106Role.cs [hunk 1]
```diff
@@ -158,14 +158,21 @@ public bool IsStalking
         /// <summary>
         /// Forces SCP-106 to use its portal, and Teleport to position.
         /// </summary>
-        /// <param name="position"> where player will be Teleport.</param>
-        /// <param name="cost"> how mush Vigor it's required and conssume.</param>
+        /// <param name="position">Where player will be Teleport.</param>
+        /// <param name="cost">How mush Vigor it's required and conssume.</param>
```
Comment (iopietro) [inline Exiled.API/Features/Roles/Scp106Role.cs:162]:
```suggestion
        /// <param name="cost">How much Vigor it's required and consumed.</param>
```

Comment (Thundermaker300) [inline Exiled.API/Features/Roles/Scp106Role.cs:162]:
```suggestion
        /// <param name="cost">The amount of vigor that will be consumed.</param>
```

File: Exiled.API/Features/Roles/Role.cs [hunk 2]
```diff
@@ -185,13 +185,21 @@ protected Role(PlayerRoleBase baseRole)
...
+            Set(newRole, reason, RoleSpawnFlags.All);
+
         /// <summary>
         /// Sets the player's <see cref="RoleTypeId"/>.
         /// </summary>
         /// <param name="newRole">The new <see cref="RoleTypeId"/> to be set.</param>
         /// <param name="reason">The <see cref="Enums.SpawnReason"/> defining why the player's role was changed.</param>
         /// <param name="spawnFlags">The <see cref="RoleSpawnFlags"/> defining player spawn logic.</param>
-        public virtual void Set(RoleTypeId newRole, SpawnReason reason = Enums.SpawnReason.ForceClass, RoleSpawnFlags spawnFlags = RoleSpawnFlags.All) =>
+        public virtual void Set(RoleTypeId newRole, SpawnReason reason, RoleSpawnFlags spawnFlags) =>
```
Comment (Thundermaker300) [inline Exiled.API/Features/Roles/Role.cs:202]: Maybe add an overload that doesn't take any params other than RoleTypeId? So plugins aren't required to input the reason property

File: Exiled.API/Features/Roles/Scp106Role.cs [hunk 3]
```diff
@@ -158,14 +158,21 @@ public bool IsStalking
...
         /// <returns>If the player will be teleport.</returns>
         public bool UsePortal(Vector3 position, float cost = 0f)
         {
+            if (Room.Get(position) is not Room room)
+            {
+                // TODO: normal message
+                throw new System.InvalidOperationException("Room doesn't found");
```
Comment (Thundermaker300) [inline Exiled.API/Features/Roles/Scp106Role.cs:169]:
```suggestion
                throw new System.InvalidOperationException("Invalid room provided.");
```

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Roles/Scp106Role.cs [hunk 4]
```diff
@@ -158,14 +158,21 @@ public bool IsStalking
         /// <summary>
         /// Forces SCP-106 to use its portal, and Teleport to position.
         /// </summary>
-        /// <param name="position"> where player will be Teleport.</param>
-        /// <param name="cost"> how mush Vigor it's required and conssume.</param>
+        /// <param name="position">Where player will be Teleport.</param>
```
Comment (iopietro) [inline Exiled.API/Features/Roles/Scp106Role.cs:161]:
```suggestion
        /// <param name="position">Where the player will be Teleported.</param>
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Roles/Role.cs (hunk 2 above)
Comment (IRacle1) [reply Exiled.API/Features/Roles/Role.cs:202]: Already, you can use  first overload `Set(RoleTypeId, SpawnReason = SpawnReason.ForceClass)` without second argument

File: Exiled.API/Features/Roles/Scp106Role.cs (hunk 1 above)
Comment (IRacle1) [reply Exiled.API/Features/Roles/Scp106Role.cs]: Already do it in the thunder review

Comment (Thundermaker300) [reply Exiled.API/Features/Roles/Scp106Role.cs]: I got this boss

Comment (Thundermaker300) [reply Exiled.API/Features/Roles/Scp106Role.cs]: Documentation is my calling

File: Exiled.API/Features/Roles/Role.cs (hunk 2 above)
Comment (Thundermaker300) [reply Exiled.API/Features/Roles/Role.cs:202]: Oh I see, my bad

File: Exiled.API/Features/Roles/Scp106Role.cs [hunk 5]
```diff
@@ -158,14 +158,20 @@ public bool IsStalking
         /// <summary>
         /// Forces SCP-106 to use its portal, and Teleport to position.
         /// </summary>
-        /// <param name="position"> where player will be Teleport.</param>
-        /// <param name="cost"> how mush Vigor it's required and conssume.</param>
+        /// <param name="position">Where the player will be Teleported.</param>
```
Comment (Thundermaker300) [inline Exiled.API/Features/Roles/Scp106Role.cs:161]:
```suggestion
        /// <param name="position">Where the player will be teleported.</param>
```
Teleported should not be capital in this case, it's not a proper noun nor the first word of a sentence.

**Review by iopietro — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**